server/trpc/router/sendEmail.ts

```diff
@@ -0,0 +1,19 @@
+import toast from "react-hot-toast";
+import { SendEmailSchema } from "../../../schema/comment";
+import sendEmail from "../../../utils/sendEmail";
```

Comment:
this may need to be updated to fix your issue propagating error from nodemailer back to the client - we should return a promise so we can Propagate the error back to the front end. in sendEmail wrap the callback function transporter.sendMail in a promise. 
```
return new Promise((resolve, reject) => {
    transporter.sendMail( ..... . etc 
```
you also need to add some logic for reject so it will pass back to the client:
`if(err){reject(new Error("Failed to send email"))}`

Reply:
try using try catch down below in sendEmailRouter and if it doesnt work wrap sendEmail in utils/sendEmail in a promise is what i would do to fix you onError in the frontend